src/lib.zig

```diff
@@ -487,6 +487,16 @@ pub fn deserialize(comptime T: type, serialized: []const u8, out: *T, allocator:
     }
 }
 
+fn mixInLength2(root: [32]u8, length: usize, out: *[32]u8) void {
+    var hasher = sha256.init(sha256.Options{});
+    hasher.update(root[0..]);
+
+    var tmp = [_]u8{0} ** 32;
+    std.mem.writeInt(@TypeOf(length), tmp[0..@sizeOf(@TypeOf(length))], length, std.builtin.Endian.little);
+    hasher.update(tmp[0..]);
+    hasher.final(out[0..]);
+}
+
```

Comment:
you could just call `mixInLength` with `tmp` as an argument.

Also it needs to have a better name.

Reply:
yes, i didn't want to convert length into 32 bytes outside , may be we should just replace `mixInLength` with this fn because i don't know why is that interface more useful

Reply:
fine, then do that please